### PR TITLE
Labels are now lexer Tokens

### PR DESCRIPTION
**Pull request by g0t00**

### CODE REVIEW

File: lib/parser/parser-base.ts [hunk 1]
```diff
@@ -363,6 +363,11 @@ export class ParserBase {
       this.consumeToken();
     }
   }
+  maybeToken(expected: OLexerToken) { // This only compares the text!
+    if (this.getToken().getLText() === expected.getLText()) {
+      this.consumeToken();
+    }
+  }
```
Comment (Nik-Sch) [inline lib/parser/parser-base.ts:370]: wouldn't it make more sense to just replace `maybeWord`?

**Comment by Nik-Sch** [comment]
In general, we might just want to remove all the functions that deal with "words" instead of tokens at this point.
Maybe escalate this PR to do just that?

File: lib/parser/parser-base.ts (hunk 1 above)
Comment (g0t00) [reply lib/parser/parser-base.ts:370]: If maybeWord is unused yes.

**Comment by g0t00** [comment]
> In general, we might just want to remove all the functions that deal with "words" instead of tokens at this point. Maybe escalate this PR to do just that?



> In general, we might just want to remove all the functions that deal with "words" instead of tokens at this point. Maybe escalate this PR to do just that?

Yes agreed. Feel free to do it soon. Otherwise I would prefer to merge this earlier and remove the word based functions later.

**Pull request merged**